### PR TITLE
Enable prometheus metrics for katib-controller

### PR DESCRIPTION
**Pull request by hougangliu**
Upgrade controller-runtime to 0.1.9 and enable  prometheus metrics for katib-controller
```
# curl http://10.0.149.224:8080/metrics
...
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 1.8476192e+07
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 1.819165e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 1.671168e+06
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 1.671168e+06
# HELP go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 7.3072888e+07
# HELP go_threads Number of OS threads created.
# TYPE go_threads gauge
go_threads 21
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 5.09
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 12
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 4.6043136e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.56568718538e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 1.41164544e+08
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes -1
...
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/717)

### CODE REVIEW

**Comment by johnugeorge** [comment]
Don't we need katib specific metrics to be exposed?

**Comment by hougangliu** [comment]
> Don't we need katib specific metrics to be exposed?

This PR only exposes controller-runtime default metrics for controller. I will submit other PRs for experiments/trials etc.

**Comment by hougangliu** [comment]
Disable UI build so that the build error will not block katib CI.
@andreyvelich will be back to fix #699

File: test/workflows/components/workflows-v1alpha2.libsonnet [hunk 1]
```diff
@@ -311,7 +311,7 @@
             $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("python-tests", pythonImage, [
               "test/scripts/v1alpha2/python-tests.sh",
             ]),  // run python tests
-            $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("run-tests", helmImage, [
+            $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("run-tests", "liuhougangxa/golang-gcloud:1.12.8-257.0.0", [
```
Comment (johnugeorge) [inline test/workflows/components/workflows-v1alpha2.libsonnet]: Why can't we use standard go image here?

Comment (hougangliu) [reply test/workflows/components/workflows-v1alpha2.libsonnet]: helmImage is an image with gcloud and golang:1.9, but now after some dependency in vendor upgraded, golang 1.9 cannot work. we need a new image with gcloud and golang 1.10 at least

Comment (johnugeorge) [reply test/workflows/components/workflows-v1alpha2.libsonnet]: How about using  kubeflow test worker image like in other repos - gcr.io/kubeflow-ci/test-worker:v20190421-fba47fe-e3b0c4 ?

Comment (hougangliu) [reply test/workflows/components/workflows-v1alpha2.libsonnet]: I will try if there are such  test worker images in gcr.io/kubeflow-ci can work

Comment (hougangliu) [reply test/workflows/components/workflows-v1alpha2.libsonnet]: gcr.io/kubeflow-ci/test-worker can work indeed. Thanks for your comment

**Pull request merged**